crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -113,6 +113,7 @@ const REGEX_CACHE_SIZE: usize = 100; // must be nonzero, otherwise will panic
 pub const NU_VARIABLE_ID: usize = 0;
 pub const IN_VARIABLE_ID: usize = 1;
 pub const ENV_VARIABLE_ID: usize = 2;
+pub const NOTHING_VARIABLE_ID: usize = 3;
```

Comment:
Are there any checks that need updating based on the note below?

Reply:
```rust
// NOTE: If you add more to this list, make sure to update the > checks based on the last in the list
```
The line below explicitly warns of checks that may depend on the range of `VarId`s reserved for those special variables that may require different actions than a regular variable.

I found this line that assumes `ENV_VARIABLE_ID` to be the last special variable. 
https://github.com/nushell/nushell/blob/e96039fb1bee747dc2529cf053e7ee00e704c79e/crates/nu-parser/src/parser.rs#L6121

With the fact that `$nothing` pretty much behaves like a boring variable/constant in `eval.rs` I am not sure if there is an edge case where this could lead to breakage.

But I wanted to make sure you have investigated the impact.

Reply:
ooh very good catch, actually i did not, let me look at these :yum:

Reply:
okey, i've fixed the only such check i could find when looking at the dozen uses of the `ENV_VARIABLE_ID` constant in c57398ad9 :+1: